woodwork/column_accessor.py

```diff
@@ -357,6 +361,62 @@ def set_semantic_tags(self, semantic_tags):
             _raise_init_error()
         self._schema._set_semantic_tags(semantic_tags)
 
+    def outliers_dict(self, low_bound=None, high_bound=None):
+        """Gets the values of a series that lay above and below specified bounds.
+
+        Args:
+            series (Series): Data for which the outliers should be determined.
+            low_bound (float, optional): The number below which outliers lay. Is inclusive.
+            high_bound (float, optional): The number above which outliers lay. Is inclusive.
+
+        Note:
+            If neither or only one of low_bound or high_bound is passed in, the bounds will be calculated
+            using the IQR method.
+
+        Returns:
+            dict[str -> float,list[number]]: a dictionary containing outlier values and their corresponding indexes.
+                The following elements will be found in the dictionary:
+
+                - low_values (list[float, int]): the values of the lower outliers
+                - high_values (list[float, int]): the values of the upper outliers
+                - low_indices (list[int]): the corresponding index values for each of the lower outliers
+                - high_indices (list[int]): the corresponding index values for each of the upper outliers
+        """
+        if self._schema is None:
+            _raise_init_error()
+
+        if self._schema.is_numeric:
+            return _get_outliers_for_column(self._series, low_bound=low_bound, high_bound=high_bound)
```

Comment:
I did intend that behavior (though I forgot to indicate that in the docstrings, so that definitely needs to be updated if we keep this behavior), but I'm not totally sold on whether this should be the behavior.

If we want to add anything, I'd probably prefer a warning over an error. I don't think we have a perfect example of this kind of behavior in the column accessor. The closest is in mutual information where we ignore incompatible columns, but we're performing the calculations for the whole table there, so it's that certain columns are missing rather than nothing happens when the method is called.

Right now I'm leaning towards adding a Warning.

Reply:
This seems like a particular instance of a larger question that I often encounter which is: how does a library (or language) prefer to represent error conditions?  I usually try to avoid concrete, primitive values such as `-1` or `None`.  The `None` built-in can be useful to represent optional arguments for which no value was provided, but I don't like it as a return value to represent some kind of failure.  That's because the `None` value is totally opaque and not descriptive.  Also, it can potentially be propagated downstream in a computation if it's not handled properly where it might have a different meaning entirely (e.g. `bool(None) is False`) or where it's just confusing and out of context.  For these reasons, I usually just prefer raising an exception and preferably a custom `Exception` subclass.  A custom subclass can be as descriptive (has its own name, error message) and specific (is only used in a very particular situation) as you want.  It also *always* has the intended effect of halting program execution until it bubbles up into a handler that is expecting it.  Would be happy to have a teams chat about this with anyone who's interested in discussing the larger question further.

Reply:
But yeah, maybe I need to examine the context of this issue more carefully.  If it's like you say, and there's some blanket operation happening for an entire table, it would make sense to not have the entire operation fail with an exception.  In that case, ideally, I would hope that there's some way to handle the exception that indicates some operation is not supported for a column and just put a `pass` in the `except` block.  It might also make sense to emit a warning like you say so the user is aware that something weird happened internally.  But, if it's the sort of thing that might actually happen pretty often on average, it might not be worth emitting a warning.

Reply:
In removing the `outliers_dict` method to just have a single `box_plot_dict` method, was able to move the `is_numeric` check to exist as a part of the statistics_utils.py `_get_box_plot_info_for_column` helper.

This check will now raise a `TypeError` for non numeric columns